doc(1.6.0): Talos Linux support (#827)
longhorn/website: content/docs/1.6.0/advanced-resources/os-distro-specific/talos-linux-support.md

---
title:  Talos Linux Support
weight: 5
---

> Before Longhorn installation on a Talos Linux cluster, specific configurations are required.
> 
> - [Installation Of System Extensions](#installation-of-system-extensions)
> - [Data Path Mounts](#data-path-mounts)
> - [References](#references)

Reviewer: ```suggestion
You must perform the following tasks before installing Longhorn on a Talos Linux cluster.
```

Reviewer: The links are not necessary given the content length.

---
title:  Talos Linux Support
weight: 5
---

Before Longhorn installation on a Talos Linux cluster, specific configurations are required.

- [Installation Of System Extensions](#installation-of-system-extensions)
- [Data Path Mounts](#data-path-mounts)
- [References](#references)

## Installation Of System Extensions

Longhorn depends on `siderolabs/iscsi-tools` and `siderolabs/util-linux-tools` system extensions, which need to be installed before installing Longhorn.

> 

Reviewer 2: Briefly explain why we need these extentions.

The most straightforward way is patching these system extensions into existing Talos Linux nodes.

```yaml
customization:
  systemExtensions:
    officialExtensions:
      - siderolabs/iscsi-tools
      - siderolabs/util-linux-tools
```

For detailed instructions, see Talos' documentation on [System Extensions](https://www.talos.dev/v1.6/talos-guides/configuration/system-extensions/) and [Boot Assets](https://www.talos.dev/v1.6/talos-guides/install/boot-assets/).

## Pod Security

Longhorn requires pod security `enforce: "privileged"`.

> 

Reviewer 2: Can we briefly explain why we need this? so users will better understand the context behind.

For detailed instructions, see Talos' documentation on [Pod Security](https://www.talos.dev/v1.6/kubernetes-guides/configuration/pod-security/).

## Data Path Mounts

Longhorn requires to mount the datapath to the kubelete container.

> 

Reviewer 2: Can we briefly explain why we need this? so users will better understand the context behind.
